src/controller.cpp

```diff
@@ -555,13 +555,17 @@ int Controller::run(const CliArgsParser& args)
 		std::cout.flush();
 	}
 	try {
-		const std::uint64_t amt = rsscache->cleanup_cache(
+		const auto unreachable_feeds = rsscache->cleanup_cache(
 				feedcontainer.get_all_feeds());
 		if (!args.silent()) {
 			std::cout << _("done.") << std::endl;
-			if (amt > 0u) {
-				std::cout << _("Unreachable feeds found, consider setting "
-						"`cleanup-on-quit yes` or run `newsboat --cleanup`")
+			if (!unreachable_feeds.empty()) {
+				for (const auto& feed : unreachable_feeds) {
+					LOG(Level::USERERROR, "Unreachable feed found: %s", feed);
+				}
+
+				std::cout << strprintf::fmt(_("%d unreachable feeds found. See "
```

Comment:
`cleanup_cache` returns a vector of feeds *and items*, so this message is misleading. Also, I think mixing URLs of feeds with URLs of items can be confusing. How about returning two separate vectors and displaying two separate counts?

Reply:
I am not sure if this is the case here.
An excerpt from my `cache.db`:

```sqlite
sqlite> select title, url, feedurl from rss_item limit 90000, 4;
title                                                         url                                                           feedurl
------------------------------------------------------------  ------------------------------------------------------------  ---------------------------
Ketchum, Idaho, Has Plenty of Available Jobs, but Workers Ca  https://www.wsj.com/articles/ketchum-idaho-has-plenty-of-ava  https://hnrss.org/frontpage
n’t Afford Housing                                            ilable-jobs-but-workers-cant-afford-housing-11625659200

Neurons Unexpectedly Encode Information in the Timing of The  https://www.quantamagazine.org/a-new-kind-of-information-cod  https://hnrss.org/frontpage
ir Firing                                                     ing-seen-in-the-human-brain-20210707/

Designing Our Serverless Engine: From Kubernetes to Nomad, F  https://www.koyeb.com/blog/the-koyeb-serverless-engine-from-  https://hnrss.org/frontpage
irecracker, and Kuma                                          kubernetes-to-nomad-firecracker-and-kuma

Game Gulf (2013)                                              https://the-magazine.org/11/game-gulf                         https://hnrss.org/frontpage
```

Since we only fetch `feedurl`s from `rss_item`, no actual items make it into this vector. In any case they would not satisfy the `NOT IN (feeds...)` criterion.

I think this is a good thing too, since the number of items would become pretty large if printed out in the user log, and printing two separate counters would go against the original goal of making the warning message less annoying.

Reply:
This means of course that the docstring for `cleanup_cache` was not correct :)

Reply:
Ah, the docstring threw me off. Thanks for fixing it!